torch/distributed/distributed_c10d.py

```diff
@@ -1218,6 +1224,8 @@ def _new_process_group_helper(
     This function is called with ``global_ranks_in_group == []`` for the default group.
     """
     global _world
+    global _nccl_subgroup_color
+    _nccl_subgroup_color += 1
```

Comment:
Can you put a comment for the +1 logic here?

Reply:
this logic was bad (and a leftover from a previous iteration).  I'm removing it in favor of a hash based color selection